api/src/test/java/org/openmrs/module/fhir2/api/dao/impl/FhirLocationDaoImplTest.java

```diff
@@ -192,4 +325,140 @@ public void getActiveAttributesByLocationAndAttributeTypeUuid_shouldReturnLocati
+		}
+		
+		sort.setOrder(SortOrderEnum.DESC);
+		
+		resultsList = new ArrayList<>(getNonNullLocationListForSorting(sort));
+		// check if the sorting is indeed correct by descending order
+		for (int i = 1; i < resultsList.size(); i++) {
+			assertThat(resultsList.get(i - 1).getStateProvince(),
+			    greaterThanOrEqualTo(resultsList.get(i).getStateProvince()));
+		}
+	}
+	
+	@Test
+	public void searchForLocations_shouldSortLocationsByPostalCodeAsRequested() {
+		SortSpec sort = new SortSpec();
+		sort.setParamName("address-postalCode");
+		sort.setOrder(SortOrderEnum.ASC);
+		
+		List<Location> resultsList = new ArrayList<>(getNonNullLocationListForSorting(sort));
+		// check if the sorting is indeed correct by ascending order
+		for (int i = 1; i < resultsList.size(); i++) {
+			assertThat(resultsList.get(i - 1).getPostalCode(), lessThanOrEqualTo(resultsList.get(i).getPostalCode()));
+		}
+		
+		sort.setOrder(SortOrderEnum.DESC);
+		
+		resultsList = new ArrayList<>(getNonNullLocationListForSorting(sort));
+		// check if the sorting is indeed correct by descending order
+		for (int i = 1; i < resultsList.size(); i++) {
+			assertThat(resultsList.get(i - 1).getPostalCode(), greaterThanOrEqualTo(resultsList.get(i).getPostalCode()));
+		}
+	}
+	
+	@Test
+	public void searchForLocations_shouldSortLocationsByCityAsRequested() {
+		SortSpec sort = new SortSpec();
+		sort.setParamName("address-city");
+		sort.setOrder(SortOrderEnum.ASC);
+		
+		List<Location> resultsList = new ArrayList<>(getNonNullLocationListForSorting(sort));
+		// check if the sorting is indeed correct by ascending order
+		for (int i = 1; i < resultsList.size(); i++) {
+			assertThat(resultsList.get(i - 1).getCityVillage(), lessThanOrEqualTo(resultsList.get(i).getCityVillage()));
+		}
+		
+		sort.setOrder(SortOrderEnum.DESC);
+		
+		resultsList = new ArrayList<>(getNonNullLocationListForSorting(sort));
+		// check if the sorting is indeed correct by descending order
+		for (int i = 1; i < resultsList.size(); i++) {
+			assertThat(resultsList.get(i - 1).getCityVillage(), greaterThanOrEqualTo(resultsList.get(i).getCityVillage()));
+		}
+	}
+	
+	private List<Location> getNonNullLocationListForSorting(SortSpec sort) {
+		Collection<Location> locations = fhirLocationDao.searchForLocations(null, null, null, null, null, null, null, sort);
+		
+		assertThat(locations, notNullValue());
+		assertThat(locations, not(empty()));
+		assertThat(locations.size(), greaterThan(1));
+		
+		List<Location> locationList = new ArrayList<>(locations);
+		// Remove people with sort parameter value null, to allow comparison while asserting.
+		switch (sort.getParamName()) {
+			case "name":
+				locationList.removeIf(p -> p.getName() == null);
+				break;
+			case "address-city":
+				locationList.removeIf(p -> p.getCityVillage() == null);
+				break;
+			case "address-state":
+				locationList.removeIf(p -> p.getStateProvince() == null);
+				break;
+			case "address-postalCode":
```

Comment:
`address-postalcode`

Reply:
Yeah I noticed that because of the coverage :P